### PR TITLE
add messages to schema

### LINKED ISSUE #1666: Add "messages" to io-package Json-Schema Definition
@GermanBluefox Please provide all details to document the new "messages" property in io-package that we can add it to the schema

e.g. https://github.com/ioBroker/ioBroker.knx/blob/master/io-package.json#L290-L342

### PR DESCRIPTION
**Pull request by foxriver76**
- closes #1666

### CODE REVIEW

File: schemas/io-package.json [hunk 1]
```diff
@@ -1100,6 +1100,62 @@
...
+            "required": ["title", "text", "level", "buttons"],
+            "properties": {
+              "condition": {
+                "description": "Message will only be shown if condition matches",
+                "additionalProperties": false,
+                "type": "object",
+                "properties": {
+                  "operand": {
+                    "description": "Operand to connect multiple rules",
+                    "enum": ["or", "and"]
```
Comment (foxriver76) [inline schemas/io-package.json:1119]: I guess these are the only two?

File: schemas/io-package.json [hunk 2]
```diff
@@ -1100,6 +1100,62 @@
...
           "maximum": 3,
           "default": 3
+        },
+        "messages": {
+          "description": "Messages can be shown on adapter installation, update and instance creation",
+          "type": "array",
+          "items": {
+            "type": "object",
+            "additionalProperties": false,
+            "required": ["title", "text", "level", "buttons"],
```
Comment (foxriver76) [inline schemas/io-package.json:1110]: What is really required?

File: schemas/io-package.json [hunk 3]
```diff
@@ -1100,6 +1100,62 @@
...
+              },
+              "linkText": {
+                "description": "If a link is defined and should be embedded in text, you can define the text here",
+                "$ref": "#/definitions/multilingual"
+              },
+              "buttons": {
+                "description": "Predefined buttons of the message window",
+                "type": "array",
+                "items": {
+                  "enum": ["agree", "cancel"]
```
Comment (foxriver76) [inline schemas/io-package.json:1154]: Are buttons a predefined selection, if so what is allowed beside agree and cancel

File: schemas/io-package.json [hunk 4]
```diff
@@ -1100,6 +1100,62 @@
...
+                "description": "Text of the shown message",
+                "$ref": "#/definitions/multilingual"
+              },
+              "link": {
+                "description": "You can define a hyperlink, which will be shown, define its text with `linkText`",
+                "type": "string"
+              },
+              "level": {
+                "description": "The importance of the message",
+                "enum": ["info", "warn", "error"]
```
Comment (foxriver76) [inline schemas/io-package.json:1144]: What is supported?

File: schemas/io-package.json (hunk 1 above)
Comment (GermanBluefox) [reply schemas/io-package.json:1119]: Or nothing. If nothing => "and"

File: schemas/io-package.json (hunk 2 above)
Comment (GermanBluefox) [reply schemas/io-package.json:1110]: - level is optional as it has default value "warn"
- title is mandatory
- text is mandatory
- link is optional
- linkText is optional and has default value "More info"

File: schemas/io-package.json (hunk 4 above)
Comment (GermanBluefox) [reply schemas/io-package.json:1144]: warn, error, info. 
That's all

File: schemas/io-package.json (hunk 3 above)
Comment (GermanBluefox) [reply schemas/io-package.json:1154]: "Buttons" is mandatory attribute, else the message will be not shown.
It has no predefined value. Possible values: "ok", "agree", "cancel"

File: schemas/io-package.json (hunk 2 above)
Comment (foxriver76) [reply schemas/io-package.json:1110]: Buttons?

Comment (GermanBluefox) [reply schemas/io-package.json:1110]: Buttons are required too

**Comment by foxriver76** [comment]
Review comments are included

**Comment by foxriver76** [comment]
for the multilingual linkText I cannot define the default value properly, so I would just leave it out

**Comment by foxriver76** [comment]
after merge we need to port to schemastore to make changes public

**Comment by GermanBluefox** [comment]
LGTM

**Pull request merged**